includes/admin/class-wc-stripe-settings-controller.php

```diff
@@ -142,9 +142,29 @@ public function admin_scripts( $hook_suffix ) {
 	 * to display the payment gateways on the WooCommerce Settings page.
 	 */
 	public static function hide_gateways_on_settings_page() {
+		$gateways_to_hide = [
+			// Hide all UPE payment methods.
+			WC_Stripe_UPE_Payment_Method::class,
+			// Hide all legacy payment methods.
+			WC_Gateway_Stripe_Alipay::class,
+			WC_Gateway_Stripe_Sepa::class,
+			WC_Gateway_Stripe_Giropay::class,
+			WC_Gateway_Stripe_Ideal::class,
+			WC_Gateway_Stripe_Bancontact::class,
+			WC_Gateway_Stripe_Eps::class,
+			WC_Gateway_Stripe_P24::class,
+			WC_Gateway_Stripe_Boleto::class,
+			WC_Gateway_Stripe_Oxxo::class,
+			WC_Gateway_Stripe_Sofort::class,
+			WC_Gateway_Stripe_Multibanco::class,
+		];
+
 		foreach ( WC()->payment_gateways->payment_gateways as $index => $payment_gateway ) {
-			if ( $payment_gateway instanceof WC_Stripe_UPE_Payment_Method ) {
-				unset( WC()->payment_gateways->payment_gateways[ $index ] );
+			foreach( $gateways_to_hide as $gateway_to_hide ) {
```

Comment:
```suggestion
			foreach ( $gateways_to_hide as $gateway_to_hide ) {
```
<img width="615" alt="Screenshot 2024-02-26 at 10 22 11 am" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/c5d02e9b-9526-4ba4-a8c3-a8af1a5aa102">

Reply:
Thanks @james-allan. I think my husky pre-commit checks are broken since upgrade PHP versions :( I'll have to look into that